packages/ncids-js/src/components/nci-autocomplete/nci-autocomplete.component.ts

```diff
@@ -0,0 +1,351 @@
+import { AutocompleteOptions } from './utils/autocomplete-options';
+import { AutocompleteAdaptor } from './utils/autocomplete-adaptor';
+
+/**
+ * NCI Autocomplete
+ * Initialize designated search inputs with augmented autocomplete markup
+ */
+
+export class NCIAutocomplete {
+	/** placeholder container for autocomplete elements */
+	protected autocompleteContainer: HTMLElement;
+	/** Url to use when fetching terms, returns an array */
+	protected adaptor: AutocompleteAdaptor;
+	/** Optional settings for the component. */
+	protected options: AutocompleteOptions;
+	/** Default settings for the component. */
+	private static optionDefaults: AutocompleteOptions = {
+		highlightMatchingText: true,
+		maxOptionsCount: 10,
+		minCharCount: 3,
+		minPlaceholderMsg: '',
+	};
+	/** Input field*/
+	protected searchField: HTMLInputElement;
+	/** Input field id used for creating other ids */
+	protected searchFieldId: string;
+	/** listbox wrapper */
+	protected listboxWrapper: HTMLDivElement;
+	/** listbox of suggestions/terms */
+	protected listbox: HTMLDivElement;
+	/** Results announcer for screenreader */
+	protected announcer: HTMLDivElement;
+
+	/** Callback for handling input changes  */
+	private inputEventListener: EventListener = () => this.handleInput();
+	/** Callback for handling keyboard controls  */
+	private keyPressEventListener: EventListener = (event: Event) =>
+		this.handleKeypress(event);
+
+	/**
+	 * Sets component properties and initializes component.
+	 *
+	 * @param {HTMLElement} autocompleteContainer container of input being created as autocomplete.
+	 * @param {URL} apiUrl url for api to be used for fetching terms
+	 * @param {AutocompleteOptions} options Autocomplete options used for component creation
+	 * @protected
+	 */
+	protected constructor(
+		autocompleteContainer: HTMLElement,
+		adaptor: AutocompleteAdaptor,
+		options?: Partial<AutocompleteOptions>
+	) {
+		this.autocompleteContainer = autocompleteContainer;
+		this.adaptor = adaptor;
+		this.options = {
+			...NCIAutocomplete.optionDefaults,
+			...options,
+		};
+		this.searchField = <HTMLInputElement>(
+			this.autocompleteContainer.querySelector('.usa-input')
+		);
+		this.searchFieldId = this.searchField.id;
+		this.listboxWrapper = document.createElement('div');
+		this.listbox = document.createElement('div');
+		this.announcer = document.createElement('div');
+		this.initialize();
+	}
+
+	/**
+	 * Instantiates this component of the given element.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @param
+	 * @param {NCIExtendedHeaderWithMegaMenuOptions} options Optional settings for component generation.
+	 * @public
+	 */
+	public static create(
+		element: HTMLElement,
+		adaptor: AutocompleteAdaptor,
```

Comment:
move adaptor into options

Reply:
`options: { autocompleteSource: AutocompleteAdapter } && Partial<AutocompleteOptions>`
make adapter a required option and options itself should be required for the create()